### ISSUE TITLE
Invite participants by name and email

### ISSUE THREAD

**Issue by PhrozenByte** [nextcloud/polls#2367]
## Describe the goal you'd like to achieve

As a poll owner, I want to provide the name of poll participants when inviting them by email, because people often fail to provide this information properly using the "Enter name and email address" dialogue.

## Describe possible solutions

The share input shouldn't just accept email addresses like `john.doe@example.com`, but also `John Doe <john.doe@example.com>`. The name ("John Doe") should then be used accordingly. The "Enter name and email address" dialogue shouldn't be shown to poll participants invited by email then.

## Additional context

I know, it's ridiculous that people fail to use the "Enter name and email address" dialogue, but it simply doesn't work. I did quite a few polls in the past with sometimes more than 100 participants, sometimes with students, sometimes with people in executive positions, most fail to provide this information when getting invited by email. Most simply skip the dialogue, some even change name and email to some random values, making it hard to later match the submissions. The number of participants getting this right is devastating...

**Comment by dartcafe** [comment]
As a workaround, you could add the participants as a contact in your contacts app. Then the name and email address will be used as prposal in the register dialog.

This way you are also able to invite them via circle or contact group.

**Comment by dartcafe** [comment]
And what about adding the email addresses as a pasted list, as a batch operation?

**Comment by PhrozenByte** [comment]
For batch operations (10+ participants or so, below I use the GUI) I usually use #1516. However, if the share input would accept multiple invitations at once, I would definitely use it :+1:

Since I often invite people only once to a single poll it's not really viable to add everyone to the contacts list first, just to remove them again. We don't use circles either.

**Comment by dartcafe** [comment]
thinking about both, the bulk import (like we added it for textoptions in the upcomming release, #2339) and adding a parser for valid email syntax. I guess I have to fight with regular expressions. ;-)

**Comment by dartcafe** [comment]
This way? 
![grafik](https://user-images.githubusercontent.com/26707476/163559322-ce2ea87d-bfb9-4ad9-9ad3-28a2709e0b7e.png)
![grafik](https://user-images.githubusercontent.com/26707476/163559329-1307b032-a6f6-4cfa-b82c-70a9aeb1723c.png)

**Comment by PhrozenByte** [comment]
Sorry for the late response. Just tested it with the latest rc1, great work! :rocket: Thanks @dartcafe for the super fast implementation :+1: 

There's just one small issue left: One of my main goals with this was to make the "Enter name and email address" dialogue optional. Since we now have both information can we switch to omitting this dialogue? Or shall I open a separate issue for this?

**Comment by dartcafe** [comment]
Yes, please add a new issue for that. But don't expect this too early, since this could raise privacy concerns, why a user cannot change name and mail address.
